src/OpenTelemetry/Trace/Internal/EvictingQueue.cs

```diff
@@ -59,6 +60,16 @@ IEnumerator IEnumerable.GetEnumerator()
             return new Enumerator(this);
         }
 
+        internal void Replace(T item, T newItem)
+        {
+            Debug.Assert(item != null, "Item must not be null");
+            Debug.Assert(newItem != null, "Item must not be null");
+
+            var index = Array.IndexOf(this.items, item);
```

Comment:
some safety check on index? what is item is not found in this.items?

Reply:
Hi @cijothomas , i was validating in the methods that used EvictingQueue.Add method. What do you suggest? Thank you for reviewing

Reply:
Just checked with this:
```csharp
if (index < 0)
{
   return;
}
```

what do you think?

Reply:
This is better!

Reply:
i'd add a unit test for this as well.

Reply:
Thanks! just added!